DevEdu.API/DevEdu.Business/Services/RatingService.cs

```diff
@@ -1,33 +1,71 @@
-﻿using DevEdu.DAL.Models;
+﻿using DevEdu.Business.Constants;
+using DevEdu.Business.Exceptions;
+using DevEdu.Business.ValidationHelpers;
+using DevEdu.DAL.Enums;
+using DevEdu.DAL.Models;
 using DevEdu.DAL.Repositories;
+using System;
 using System.Collections.Generic;
 
 namespace DevEdu.Business.Services
 {
     public class RatingService : IRatingService
     {
         private readonly IRatingRepository _repository;
+        private readonly IRatingValidationHelper _ratingValidationHelper;
+        private readonly IGroupValidationHelper _groupValidationHelper;
+        private readonly IUserValidationHelper _userValidationHelper;
 
-        public RatingService(IRatingRepository repository)
+        public RatingService(IRatingRepository repository, IRatingValidationHelper ratingValidationHelper,
+            IGroupValidationHelper groupValidationHelper, IUserValidationHelper userValidationHelper)
         {
             _repository = repository;
+            _ratingValidationHelper = ratingValidationHelper;
+            _groupValidationHelper = groupValidationHelper;
+            _userValidationHelper = userValidationHelper;
         }
 
-        public int AddStudentRating(StudentRatingDto studentRatingDto) => _repository.AddStudentRating(studentRatingDto);
+        public int AddStudentRating(StudentRatingDto studentRatingDto, int authorUserId)
+        {
+            _groupValidationHelper.CheckGroupExistence(studentRatingDto.Group.Id);
+            _groupValidationHelper.CheckTeacherBelongToGroup(studentRatingDto.Group.Id, Convert.ToInt32(authorUserId), Role.Teacher);
+            _userValidationHelper.CheckUserExistence(studentRatingDto.User.Id);
+            _groupValidationHelper.CheckUserBelongToGroup(studentRatingDto.Group.Id, studentRatingDto.User.Id, Role.Student);
+            return _repository.AddStudentRating(studentRatingDto);
+        }
 
-        public void DeleteStudentRating(int id) => _repository.DeleteStudentRating(id);
+        public void DeleteStudentRating(int id, int authorUserId)
+        {
+            var dto = GetStudentRatingById(id);
+            _groupValidationHelper.CheckTeacherBelongToGroup(dto.Group.Id, Convert.ToInt32(authorUserId), Role.Teacher);
+            _repository.DeleteStudentRating(id);
+        }
 
         public List<StudentRatingDto> GetAllStudentRatings() => _repository.SelectAllStudentRatings();
 
-        public StudentRatingDto GetStudentRatingById(int id) => _repository.SelectStudentRatingById(id);
+        public StudentRatingDto GetStudentRatingById(int id)
+        {
+            var dto = _repository.SelectStudentRatingById(id);
+            if (dto == default)
+            {
+                throw new EntityNotFoundException(string.Format(ServiceMessages.EntityNotFoundMessage, nameof(dto), id));
+            }
+            return dto;
+        }
 
         public List<StudentRatingDto> GetStudentRatingByUserId(int userId) => _repository.SelectStudentRatingByUserId(userId);

```

Comment:
so you have a lot of checks when someone tries to add, delete or get a rating by id
but for methods with lists the gates are open X)

I'd like to see some authorization here)